docs/src/extend/custom-parsers.md

```diff
@@ -74,8 +63,45 @@ interface Token {
 
 The range indexes of all tokens and comments must not overlap with the range of other tokens and comments.
 
-### The `Literal` node:
+### The `Literal` Node
 
 The `Literal` node must have `raw` property.
 
 * `raw` (`string`) is the source code of this literal. This is the same as `code.slice(node.range[0], node.range[1])`.
+
+## Packaging a Custom Parser
+
+TODO: Add info on turning into a package
+
+## Example
+
+For a complex example of a custom parser, refer to the [`@typescript-eslint/parser`](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/parser) source code.
+
+A simple custom parser that logs `"foo"` to the console when it processes a node:
+
+```javascript
+// awesome-custom-parser.js
+var espree = require("espree");
+exports.parseForESLint = function(code, options) {
```

Comment:
question for reviewers: this syntax is a little different from what i'm used to seeing with Node.js. 

to my understanding, this is equivalent to the following:

```js
function parseForESLint(code, options){
// function definition
}

module.exports = { parseForESLint }
```

i *think* this `module.exports = ...` pattern is more common.

would we be open to adjusting this example to use the `module.exports = ...` pattern?

Reply:
You're correct, the two are equivalent. No objections to using the pattern you suggest.

Reply:
adding